gadgets/snapshot_socket: Missing tests

The [snapshot_socket](https://github.com/inspektor-gadget/inspektor-gadget/tree/main/gadgets/snapshot_socket) gadget is missing the tests. We can port them from the built-in gadget (https://github.com/inspektor-gadget/inspektor-gadget/blob/main/integration/k8s/snapshot_socket_test.go)

@mauriciovasquezbernal The tests are failing due to timeout. Can environment setup cause this problems?
`IG_PATH=/home/anuj_agrawal/inspektor-gadget-lfx/inspektor-gadget/inspektor-gadget/kubectl-gadget IG_RUNTIME=kubernetes go test -v
=== RUN   TestSnapshotSocket
    utils.go:178: Seed used: 1723445360569292488
    command.go:92: Run command(Create test namespace):
    command.go:94: Command returned(Create test namespace):
        
        namespace/test-snapshot-socket-1636736912831736689 created
        NAME      SECRETS   AGE
        default   0         0s
        
    command.go:111: Start command(Run test-snapshot-socket)
    command.go:92: Run command(SleepForSeconds):
    command.go:94: Command returned(SleepForSeconds):
        
        
    command.go:92: Run command(WaitForTestPod):
    command.go:94: Command returned(WaitForTestPod):
        
        pod/test-snapshot-socket condition met
        
    command.go:128: Stop command(Run test-snapshot-socket)
    command.go:130: Command returned(Run test-snapshot-socket):
        
        pod/test-snapshot-socket created
        
    command.go:92: Run command(Run_snapshot_socket):
panic: test timed out after 10m0s
running tests:
        TestSnapshotSocket (10m0s)

goroutine 30 [running]:
testing.(*M).startAlarm.func1()
        /home/anuj_agrawal/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.linux-amd64/src/testing/testing.go:2366 +0x385
created by time.goFunc
        /home/anuj_agrawal/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.linux-amd64/src/time/sleep.go:177 +0x2d
`

--- a/gadgets/snapshot_socket/test/integration/snapshot_socket_test.go
+++ b/gadgets/snapshot_socket/test/integration/snapshot_socket_test.go
@@ -1,0 +1,119 @@
+// Copyright 2019-2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	gadgettesting "github.com/inspektor-gadget/inspektor-gadget/gadgets/testing"
+	igtesting "github.com/inspektor-gadget/inspektor-gadget/pkg/testing"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/containers"
+	igrunner "github.com/inspektor-gadget/inspektor-gadget/pkg/testing/ig"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/match"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/utils"
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+)
+
+type snapshotSocketEntry struct {
+	eventtypes.CommonData
+
+	NetNsID     uint64 `json:"netns_id"`
+	InodeNumber uint64 `json:"ino"`
+
+	SrcEndpoint utils.L4Endpoint `json:"src"`
+	DstEndpoint utils.L4Endpoint `json:"dst"`
+	Status      uint64           `json:"status"`
+}
+
+func TestSnapshotSocket(t *testing.T) {
+	gadgettesting.RequireEnvironmentVariables(t)
+	utils.InitTest(t)
+
+	containerFactory, err := containers.NewContainerFactory(utils.Runtime)
+	require.NoError(t, err, "new container factory")
+	containerName := "test-snapshot-socket"
+	containerImage := "docker.io/library/busybox:latest"
+
+	var ns string
+	containerOpts := []containers.ContainerOption{containers.WithContainerImage(containerImage)}
+
+	if utils.CurrentTestComponent == utils.KubectlGadgetTestComponent {
+		ns = utils.GenerateTestNamespaceName(t, "test-snapshot-socket")
+		containerOpts = append(containerOpts, containers.WithContainerNamespace(ns))
+	}
+
+	testContainer := containerFactory.NewContainer(
+		containerName,
+		"nc -l 0.0.0.0 -p 9090",
+		containerOpts...,
+	)
+
+	testContainer.Start(t)
+	t.Cleanup(func() {
+		testContainer.Stop(t)
+	})
+
+	var runnerOpts []igrunner.Option
+	var testingOpts []igtesting.Option
+	commonDataOpts := []utils.CommonDataOption{utils.WithContainerImageName(containerImage), utils.WithContainerID(testContainer.ID())}
+
+	// TODO: timeout shouldn't be required
+	switch utils.CurrentTestComponent {
+	case utils.IgLocalTestComponent:
+		runnerOpts = append(runnerOpts, igrunner.WithFlags(fmt.Sprintf("-r=%s", utils.Runtime), "--timeout=1"))
+	case utils.KubectlGadgetTestComponent:
+		runnerOpts = append(runnerOpts, igrunner.WithFlags(fmt.Sprintf("-n=%s", ns), "--timeout=1"))
+		testingOpts = append(testingOpts, igtesting.WithCbBeforeCleanup(utils.PrintLogsFn(ns)))
+		commonDataOpts = append(commonDataOpts, utils.WithK8sNamespace(ns))
+	}
+
+	runnerOpts = append(runnerOpts, igrunner.WithValidateOutput(
+		func(t *testing.T, output string) {
+			expectedEntry := &snapshotSocketEntry{
+				CommonData: utils.BuildCommonData(containerName, commonDataOpts...),
+				SrcEndpoint: utils.L4Endpoint{
+					Addr:    "0.0.0.0",
+					Version: 4,
+					Port:    9090,
+					Proto:   "TCP",
+				},
+				DstEndpoint: utils.L4Endpoint{
+					Addr:    "0.0.0.0",
+					Version: 4,
+					Port:    0,
+					Proto:   "TCP",
+				},
+				Status:      0,
+				NetNsID:     utils.NormalizedInt,
+				InodeNumber: utils.NormalizedInt,
+			}
+
+			normalize := func(e *snapshotSocketEntry) {
+				utils.NormalizeCommonData(&e.CommonData)
+				utils.NormalizeInt(&e.NetNsID)
+				utils.NormalizeInt(&e.InodeNumber)
+			}
+
+			match.MatchEntries(t, match.JSONSingleArrayMode, output, normalize, expectedEntry)
+		},
+	))
+
+	snapshotSocketCmd := igrunner.New("snapshot_socket", runnerOpts...)
+	igtesting.RunTestSteps([]igtesting.TestStep{utils.Sleep(5 * time.Second), snapshotSocketCmd}, t, testingOpts...)
+}

